src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerRegenerateCourtOrders.java

```diff
@@ -109,31 +103,26 @@ public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(
             generateFinalOrder.removeExistingAndGenerateNewFinalOrderGrantedDoc(details);
         }
 
-        CaseDetails<CaseData, State> updatedDetails = null;
         if (isNotEmpty(caseData.getConditionalOrder().getCertificateOfEntitlementDocument())) {
             log.info("Regenerating certificate of entitlement document for Case Id: {}", details.getId());
-            generateCertificateOfEntitlement.removeExistingAndGenerateNewCertificateOfEntitlementCoverLetters(details);
-            updatedDetails = caseTasks(generateCertificateOfEntitlement).run(details);
-        }
 
-        if (null != updatedDetails) {
-            return AboutToStartOrSubmitResponse.<CaseData, State>builder()
-                .data(updatedDetails.getData())
-                .build();
-        } else {
-            log.info("Certificate of entitlement doesn't exist for Case Id: {}", details.getId());
-            return AboutToStartOrSubmitResponse.<CaseData, State>builder()
-                .data(caseData)
-                .build();
-        }
-    }
+            documentGenerationUtil.generateCertificateOfEntitlement(details);
 
-    public SubmittedCallbackResponse submitted(CaseDetails<CaseData, State> details,
-                                               CaseDetails<CaseData, State> beforeDetails) {
-        log.info("Caseworker regenerate court orders submitted callback invoked for case id: {}", details.getId());
+            log.info("Completed generating certificate of entitlement pdf for CaseID: {}", details.getId());
+            documentGenerationUtil.removeExistingDocuments(
+                    caseData,
+                    List.of(CERTIFICATE_OF_ENTITLEMENT_COVER_LETTER_APP1,
+                            CERTIFICATE_OF_ENTITLEMENT_COVER_LETTER_APP2));
+        }
 
-        final CaseData caseData = details.getData();
         notificationDispatcher.send(regenerateCourtOrdersNotification, caseData, details.getId());
-        return SubmittedCallbackResponse.builder().build();
+
+        documentGenerationUtil.removeExistingDocuments(
+                caseData,
+                List.of(CERTIFICATE_OF_ENTITLEMENT));
+
+        return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+                .data(details.getData())
```

Comment:
it might be easier to reference `caseData` here - I'm sure it's the same instance in `details.getData()` but it's a bit confusing to go from one reference to the other

Reply:
Changed it to `caseData` as suggested